src/lookup_key.rs

```diff
@@ -504,7 +504,7 @@ impl PathItem {
 /// wrapper around `getattr` that returns `Ok(None)` for attribute errors, but returns other errors
 /// We don't check `try_from_attributes` because that check was performed on the top level object before we got here
 fn py_get_attrs<'py>(obj: &Bound<'py, PyAny>, attr_name: &Py<PyString>) -> PyResult<Option<Bound<'py, PyAny>>> {
-    match obj.getattr(attr_name.extract::<&PyString>(obj.py())?) {
```

Comment:
Why were we able to make this simplification?

Sometimes I get a bit lost in a complicated `pyo3` transformation like the one above - why did we need the `obj.py()` as an argument to `extract` originally?

Reply:
`.getattr` takes anything which is convertible to `Py<PyString>`, so the `.extract()` dance was just irrelevant. The upgrade flagged this because `&PyString` as a type is now removed.

`.extract()` needed a `py` argument because `attr_name` is `Py<PyString>` which doesn't have a type-level connection to the GIL (unlike `Bound<'py, T>>`), so we have to pass it in.